### PR TITLE
Upgrade react-native-compat to avoid PropTypes warnings

### PR DESCRIPTION
**Pull request by atticoos**
One step towards https://github.com/storybooks/storybook/issues/1540 by updating the `react-native-compat` dependency to move off of `React.PropTypes` (https://github.com/ajwhite/react-native-compat/pull/1). There still needs to be a greater sweep to remove usages of `React.PropTypes` -- users will likely still see this warning on their devices.

## What I did
Upgrades to `react-native-compat@v0.0.4`, which uses `prop-types`.

## How to test
I manually pulled down `react-native-compat` and compared the old vs new version to confirm the warning goes away. However, as mentioned, that may not entirely fix the warnings throughout Storybook [we still have occurrences using it](https://github.com/storybooks/storybook/blob/master/app/react-native/src/preview/components/StoryListView/index.js#L1)).

### CODE REVIEW

File: app/react-native/package.json [hunk 1]
```diff
@@ -56,7 +56,7 @@
     "json-loader": "^0.5.4",
     "json5": "^0.5.1",
     "postcss-loader": "^2.0.5",
-    "react-native-compat": "0.0.2",
+    "react-native-compat": "0.0.4",
```
Comment (danielduan) [inline app/react-native/package.json]: do you plan on versioning according to semvar? we could do `~0.0.4` so we don't have to do a PR for every minor fix.

**Review by danielduan — APPROVED** [review]

File: app/react-native/package.json (hunk 1 above)
Comment (ajwhite) [inline app/react-native/package.json]: Yes - this is currently in a prerelease stage so we'd need to upgrade to `1.x` for precedence options.

Let me update `react-native-compat` to `1.0.0` so we can take advantage of `~`

Comment (danielduan) [inline app/react-native/package.json]: Think it really depends on how you plan to maintain it. Automatic updates should not be breaking changes. Typically `^` minor version updates are feature improvements, `~` are bug fixes. If you follow the same convention, `^` should be fine.

Comment (ajwhite) [inline app/react-native/package.json]: The project is semver compliant 👍

**Pull request merged**